src/eos-speedwagon.c

```diff
@@ -43,6 +43,78 @@ typedef struct {
 GType eos_speedwagon_window_get_type (void) G_GNUC_CONST;
 G_DEFINE_TYPE (EosSpeedwagonWindow, eos_speedwagon_window, GTK_TYPE_APPLICATION_WINDOW)
 
+static gchar *
+get_splash_background_path (GDesktopAppInfo *app_info)
+{
+  gchar *bg_path = NULL;
+
+  if (g_desktop_app_info_has_key (app_info,
+                                  SPLASH_BACKGROUND_DESKTOP_KEY))
+    {
+      gchar *bg_name =
+        g_desktop_app_info_get_string (app_info,
+                                       SPLASH_BACKGROUND_DESKTOP_KEY);
+
+      /* rewrite absolute paths */
+      if (g_path_is_absolute (bg_name))
```

Comment:
Instead of always rewriting absolute paths, maybe it would be a good idea to check first if it exists and return early if it does. I don't see us really needing that functionality, but it seems more robust if you're in a situation where speedwagon is updated but shell-content isn't. E.g.
```
if (g_path_is_absolute (bg_name))
  {
    if (g_file_test (bg_name, G_FILE_TEST_EXISTS))
      return g_strdup (bg_name);
    /* continue with basename logic */
  }
```
Although, I guess this is what @magcius was objecting to.